inject/src/main/java/io/micronaut/inject/qualifiers/MatchArgumentQualifier.java

```diff
@@ -59,17 +59,43 @@ public static <T> MatchArgumentQualifier<T> ofArgument(Argument<?> argument) {
         );
     }
 
-    public static <T> MatchArgumentQualifier<T> ofSuperVariable(Class<T> beanType, Argument<?> argument) {
-        Argument<?> superArgument = Argument.ofTypeVariable(argument.getType(), null, argument.getAnnotationMetadata(), argument.getTypeParameters());
+    /**
+     * Finds matches of a type with a covariant generic type (types that extend the type or are equal to it).
+     * The generic argument is assignable from the candidate generic type.
+     * Use-cases are generic deserializers, readers.
+     * Java example:
+     * MyReader[ArrayList[String]] candidate = ...;
+     * MyReader[? extends List[String]] aMatch = candidate;
```

Comment:
can you add `@since` here and wrap the code snippets in `<pre><code>` tags

Reply:
Add the code tag, since is not needed because the class is new in this version